src/magic.cpp

```diff
@@ -860,6 +912,224 @@ int known_magic::time_to_learn_spell( const player &p, spell_id sp ) const
+        width = std::max( width, sp->name().length() );
+    }
+    return width;
+}
+
+class spellcasting_callback : public uilist_callback
+{
+    private:
+        std::vector<spell *> known_spells;
+        void draw_spell_info( const spell &sp, const uilist *menu );
+    public:
+        bool casting_ignore;
+
+        spellcasting_callback( std::vector<spell *> &spells,
+                               bool casting_ignore ) : known_spells( spells ),
+            casting_ignore( casting_ignore ) {};
+        bool key( const input_context &, const input_event &event, int /*entnum*/,
+                  uilist * /*menu*/ ) override {
+            if( event.get_first_input() == 'I' ) {
+                casting_ignore = !casting_ignore;
+                return true;
+            }
+            return false;
+        }
+
+        void select( int entnum, uilist *menu ) override {
+            mvwputch( menu->window, 0, menu->w_width - menu->pad_right, c_magenta, LINE_OXXX );
+            mvwputch( menu->window, menu->w_height - 1, menu->w_width - menu->pad_right, c_magenta, LINE_XXOX );
+            for( int i = 1; i < menu->w_height - 1; i++ ) {
+                mvwputch( menu->window, i, menu->w_width - menu->pad_right, c_magenta, LINE_XOXO );
+            }
+            std::string ignore_string = "[I] " + casting_ignore ? _( "Ignore Distractions" ) :
+                                        _( "Popup Distractions" );
+            mvwprintz( menu->window, 0, menu->w_width - menu->pad_right + 2,
+                       casting_ignore ? c_red : c_light_green, ignore_string );
+            draw_spell_info( *known_spells[entnum], menu );
+        }
+};
+
+static std::string moves_to_string( const int moves )
+{
+    const int turns = moves / 100;
+    if( moves < 200 ) {
+        return _( string_format( "%d %s", moves, "moves" ) );
+    } else if( moves < to_moves<int>( 2_minutes ) ) {
+        return _( string_format( "%d %s", turns, "turns" ) );
+    } else if( moves < to_moves<int>( 2_hours ) ) {
+        return _( string_format( "%d %s", to_minutes<int>( turns * 1_turns ), "minutes" ) );
+    } else {
+        return _( string_format( "%d %s", to_hours<int>( turns * 1_turns ), "hours" ) );
+    }
+}
+
+void spellcasting_callback::draw_spell_info( const spell &sp, const uilist *menu )
+{
+    const int h_offset = menu->w_width - menu->pad_right + 1;
+    // includes spaces on either side for readability
+    const int info_width = menu->pad_right - 4;
+    const int h_col1 = h_offset + 1;
+    const int h_col2 = h_offset + ( info_width / 2 );
+    const catacurses::window w_menu = menu->window;
+    // various pieces of spell data mean different things depending on the effect of the spell
+    const std::string fx = sp.effect();
+    int line = 1;
+    line += fold_and_print( w_menu, line++, h_col1, info_width, c_light_gray, sp.description() );
+
+    print_colored_text( w_menu, line, h_col1, c_light_gray, c_light_gray,
```

Comment:
I don't think you can use constant color defines directly here - `print_colored_text` expects `nc_color &` as fourth parameter and can change them.